conans/model/build_info.py

```diff
@@ -333,7 +348,7 @@ def merge_lists(seq1, seq2):
         self.frameworkdirs = merge_lists(self.frameworkdirs, dep_cpp_info.framework_paths)
         self.libs = merge_lists(self.libs, dep_cpp_info.libs)
         self.frameworks = merge_lists(self.frameworks, dep_cpp_info.frameworks)
-        self.build_modules = merge_lists(self.build_modules, dep_cpp_info.build_modules_paths)
+        self.build_modules = merge_dicts(self.build_modules, dep_cpp_info.build_modules_paths)
```

Comment:
The thing is that the generators that provide 1 file per dependency ``cmake_find_package(_multi)``, ``MSBuildDeps``, which are the future, do not need an aggregated dictionary at all. Every dependency file will declare its own modules.

Maybe we could provide the new build-modules-per-generator for those generators only, and keep the old ones for the others (so we also don't break)? I can see this might be an ugly UI, but just in case, this could be an idea.